services/thumbnails/pkg/thumbnail/processor.go

```diff
@@ -0,0 +1,51 @@
+package thumbnail
+
+import (
+	"image"
+	"strings"
+
+	"github.com/disintegration/imaging"
+)
+
+// Processor processes the thumbnail by applying different transformations to it.
+type Processor interface {
+	ID() string
+	Process(img image.Image, width, height int, filter imaging.ResampleFilter) *image.NRGBA
+}
+
+// DefinableProcessor is the most simple processor, it holds a replaceable image converter function.
+type DefinableProcessor struct {
+	Slug      string
+	Converter func(img image.Image, width, height int, filter imaging.ResampleFilter) *image.NRGBA
+}
+
+// ID returns the processor identification.
+func (p DefinableProcessor) ID() string { return p.Slug }
+
+// Process transforms the given image.
+func (p DefinableProcessor) Process(img image.Image, width, height int, filter imaging.ResampleFilter) *image.NRGBA {
+	return p.Converter(img, width, height, filter)
+}
+
+// ProcessorFor returns a matching Processor
+func ProcessorFor(id, fileType string) (Processor, error) {
```

Comment:
Imho this should return a `DefinableProcessor` (return structs, expect interfaces) But this is very nitpicky I admit